awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGenerator.java

```diff
@@ -392,6 +417,97 @@ private static void setRemoteResourceTypeAndIdentifier(SpanData span, Attributes
     }
   }
 
+  /**
+   * RemoteResourceIdentifier is populated with rule <code>
+   *     ^[db.name|]?{address} [|{port}]?
+   * </code>
+   *
+   * <pre>
+   * {address} attribute is retrieved in priority order:
+   * - {@link SemanticAttributes#SERVER_ADDRESS},
+   * - {@link SemanticAttributes#NET_PEER_NAME},
+   * - {@link SemanticAttributes#SERVER_SOCKET_ADDRESS}
+   * - {@link SemanticAttributes#DB_CONNECTION_STRING}-Hostname
+   * </pre>
+   *
+   * <pre>
+   * {port} attribute is retrieved in priority order:
+   * - {@link SemanticAttributes#SERVER_PORT},
+   * - {@link SemanticAttributes#NET_PEER_PORT},
+   * - {@link SemanticAttributes#SERVER_SOCKET_PORT}
+   * - {@link SemanticAttributes#DB_CONNECTION_STRING}-Port
+   * </pre>
+   *
+   * If address is not present, neither RemoteResourceType nor RemoteResourceIdentifier will be
+   * provided. If any attribute contains | or ^ , they will be replaced with ^| or ^^
```

Comment:
Not blocking: This comment belong on `setRemoteResourceTypeAndIdentifier`, as a) it applies to all remote resources and b) this method has nothing to do with setting resource type.

The first part of this comment (set both or neither) is already present in the method-level comment, the second part (delimiter escaping) is absent. Can you migrate the second part to the comment on `setRemoteResourceTypeAndIdentifier`?